engine/evalOp.cc

```diff
@@ -488,6 +491,26 @@ namespace minsky
   template <>
   double EvalOp<OperationType::frac>::d2(double x1, double x2) const
   {return 0;}
+  
+  template <>
+  double EvalOp<OperationType::percent>::evaluate(double in1, double in2) const
+  {return 100*in1;}
+  template <>
+  double EvalOp<OperationType::percent>::d1(double x1, double x2) const
+  {return 0;}
+  template <>
+  double EvalOp<OperationType::percent>::d2(double x1, double x2) const
+  {return 0;}
+  
+  template <>
+  double EvalOp<OperationType::fact>::evaluate(double in1, double in2) const
+  {return boost::math::tgamma(in1+1);}
+  template <>
+  double EvalOp<OperationType::fact>::d1(double x1, double x2) const
+  {return boost::math::digamma(x1+1)*boost::math::tgamma(x1+1);}
```

Comment:
I was going to say use the libm gamma function, but I don't think libm contains a digamma function.

Reply:
I didn't see one either, which I was I chose boost. In general, I try to use the std library first, but when a functionality/implementation is unavailable, I go shopping elsewhere.